### PR TITLE
Chat fixes w/ inbox filter / new chat

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers i'm doing this quickly since master is kinda messed up
- [x] typing into filter will keep the first item selected. we should have been doing this all along.
- [x] mobile inbox tapping messed

- [ ] enter key moves input to input. I'm deferring this until monday. requires more plumbing and isn't a show stopper

i'm also likely going ot merge this w/o thumbs. we can fix it up later

### CODE REVIEW

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: shared/actions/chat2/index.js [hunk 1]
```diff
@@ -578,11 +578,7 @@ const loadMoreMessages = (
 
   if (action.type === Chat2Gen.setPendingConversationUsers) {
     if (state.chat2.pendingSelected) {
-      const toFind = I.Set(action.payload.users.concat([state.config.username]))
-      key = state.chat2.metaMap.findKey(meta =>
-        // Ignore the order of participants
-        meta.participants.toSet().equals(toFind)
-      )
+      key = Constants.findConversationFromParticipants(state, I.Set(action.payload.users))
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: pull into helper and only select ad hoc convos

File: shared/actions/chat2/index.js [hunk 2]
```diff
@@ -753,7 +749,9 @@ const previewConversation = (action: Chat2Gen.SelectConversationPayload) =>
     : null
 
 const clearInboxFilter = (action: Chat2Gen.SelectConversationPayload) =>
-  action.payload.fromFilter ? undefined : Saga.put(Chat2Gen.createSetInboxFilter({filter: ''}))
+  action.payload.reason === 'inboxFilterArrow' || action.payload.reason === 'inboxFilterChanged'
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: leave inbox filter if we're moving around it due to typing

File: shared/actions/chat2/index.js [hunk 3]
```diff
@@ -942,8 +940,6 @@ const sendToPendingConversationSuccess = (
     Saga.put(Chat2Gen.createExitSearch()),
     // Clear the dummy messages from the pending conversation
     Saga.put(Chat2Gen.createClearPendingConversation()),
-    // Exit pending mode
-    Saga.put(Chat2Gen.createSetPendingMode({pendingMode: 'none'})),
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: this is a side effect of exitSearch now

File: shared/actions/chat2/index.js [hunk 4]
```diff
@@ -1134,10 +1130,18 @@ const selectTheNewestConversation = (
...
+    state.chat2.pendingConversationUsers
+  )
   return Saga.sequentially([
     Saga.put(SearchGen.createClearSearchResults({searchKey: 'chatSearch'})),
     Saga.put(SearchGen.createSetUserInputItems({searchKey: 'chatSearch', searchResults: []})),
     Saga.put(Chat2Gen.createSetPendingConversationUsers({fromSearch: true, users: []})),
+    Saga.put(Chat2Gen.createSetPendingMode({pendingMode: 'none'})),
+    ...(action.payload.canceled
+      ? []
+      : [Saga.put(Chat2Gen.createSelectConversation({conversationIDKey, reason: 'startFoundExisting'}))]),
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: if we exit cause of pressing enter twice, lets just select the convo

File: shared/actions/chat2/index.js [hunk 5]
```diff
@@ -1523,13 +1527,6 @@ const markThreadAsRead = (
     | NavigateActions,
   state: TypedState
 ) => {
-  if (action.type === Chat2Gen.selectConversation) {
```
Comment (chrisnojima) [inline shared/actions/chat2/index.js]: there is no fromUser anymore

File: shared/chat/inbox/container/index.js [hunk 6]
```diff
@@ -125,6 +128,16 @@ export default compose(
           nextProps.onUntrustedInboxVisible(toUnbox)
         }
       }
+
+      // keep first item selected if filter changes
+      if (!isMobile) {
```
Comment (chrisnojima) [inline shared/chat/inbox/container/index.js]: as you type the filter always keep the first thing selected, unless you arrow around

File: shared/chat/inbox/row/chat-filter-row.js [hunk 7]
```diff
@@ -155,7 +155,7 @@ const loadingContainer = {
 const styleContainer = {
   ...globalStyles.flexBoxRow,
   alignItems: 'center',
-  backgroundColor: globalColors.blue5,
+  backgroundColor: isMobile ? globalColors.fastBlank : globalColors.blue5,
```
Comment (chrisnojima) [inline shared/chat/inbox/row/chat-filter-row.js]: this is white on mobile

File: shared/constants/chat2/index.js [hunk 8]
```diff
@@ -9,7 +9,6 @@ import {isMobile} from '../platform'
 
 export const makeState: I.RecordFactory<Types._State> = I.Record({
   badgeMap: I.Map(),
-  beforeSearchSelectedConversation: null,
```
Comment (chrisnojima) [inline shared/constants/chat2/index.js]: wasn't actually used

File: shared/search/helpers.js [hunk 9]
```diff
@@ -33,7 +33,9 @@ type OutProps = {
 }
 */
 const clearSearchHoc = withHandlers({
-  onClearSearch: ({onExitSearch}) => () => onExitSearch(),
+  // use existing onClearSearch if exists. TODO change how this whole thing works. so confusing
```
Comment (chrisnojima) [inline shared/search/helpers.js]: how search works is very convoluted. i think we can do something simpler but lets leave that for later. for now if you pass in an onClearSearch it won't blow it away

File: shared/search/helpers.js [hunk 10]
```diff
@@ -33,7 +33,9 @@ type OutProps = {
...
+  onClearSearch: ({onExitSearch, onClearSearch}) =>
+    onClearSearch ? () => onClearSearch() : () => onExitSearch(),
```
Comment (buoyad) [inline shared/search/helpers.js]: do you need these closures?

**Pull request merged**